vnext/Microsoft.ReactNative/Views/SwitchViewManager.cpp

```diff
@@ -42,12 +42,58 @@ class SwitchShadowNode : public ShadowNodeBase {
   winrt::Microsoft::ReactNative::JSValue m_onTrackColor;
 };
 
+// sets the resources to SolidColorBrushes if the user sets the color before runtime, this lets the user change the
+// color of the SolidColorBrush at runtime and keeps the toggleswitches who did not change colors on the default
+// ThemeResource
+void setBrushes(
+    const winrt::Microsoft::ReactNative::JSValueObject &props,
+    const xaml::Controls::ToggleSwitch toggleSwitch) {
+  for (const auto &pair : props) {
+    const std::string &propertyName = pair.first;
+    const auto &propertyValue = pair.second;
+    if (propertyName == "thumbTintColor") {
+      toggleSwitch.Resources().Insert(winrt::box_value(L"ToggleSwitchKnobFillOff"), xaml::Media::SolidColorBrush());
+      toggleSwitch.Resources().Insert(
+          winrt::box_value(L"ToggleSwitchKnobFillOffPointerOver"), xaml::Media::SolidColorBrush());
+      toggleSwitch.Resources().Insert(
+          winrt::box_value(L"ToggleSwitchKnobFillOffPressed"), xaml::Media::SolidColorBrush());
+      toggleSwitch.Resources().Insert(
+          winrt::box_value(L"ToggleSwitchKnobFillOffDisabled"), xaml::Media::SolidColorBrush());
+      toggleSwitch.Resources().Insert(
+          winrt::box_value(L"ToggleSwitchKnobFillOnPointerOver"), xaml::Media::SolidColorBrush());
+      toggleSwitch.Resources().Insert(
+          winrt::box_value(L"ToggleSwitchKnobFillOnDisabled"), xaml::Media::SolidColorBrush());
+      toggleSwitch.Resources().Insert(winrt::box_value(L"ToggleSwitchKnobFillOn"), xaml::Media::SolidColorBrush());
+      toggleSwitch.Resources().Insert(
+          winrt::box_value(L"ToggleSwitchKnobFillOnPressed"), xaml::Media::SolidColorBrush());
```

Comment:
this gets pretty verbose, consider factoring out to a function that takes a wstring_view and the toggleSwitch and does the insert with an empty brush

Reply:
or just use UpdateResourcesBrush

Reply:
switched to use UpdateResoucesBrush!